x-pack/plugin/spatial/build.gradle

```diff
@@ -15,6 +15,7 @@ dependencies {
   testImplementation(testArtifact(project(xpackModule('core'))))
   yamlRestTestImplementation(testArtifact(project(xpackModule('core'))))
   api project(path: ':modules:geo')
+  api project(path: xpackModule('vector-tile'))
```

Comment:
Is this enough and ok?

Reply:
If you are asking from the build POV I think @pugnascotia or @breskeby might be able to assist here

Reply:
Hey @elastic/es-delivery, do you have any feedback about the change above? We need to access a factory that lives in the vector-tile module from the spatial module.

Reply:
I defer to @breskeby here. We mostly have cross-plugin deps for test purposes, as far as I can tell, but this is probably OK. We could move it to `core`, but I don't if that would be worth it?

Reply:
I guess it would make more sense to move vector-tiles module to spatial as it is currently just adding the vector tile end point. It was originally included in the spatial module but we decided to move it its own module after discussing with Ryan. Maybe it wasn't the right decision?

Reply:
from the build perspective this is fine. I don't have much background in how comfortable we are with cross plugin dependencies though.

Reply:
I think this is all that's needed. These (despite the folder naming) are modules, not plugins, so there shouldn't be anything else needed here since that module is always guaranteed to be installed. I don't have strong opinions on how these two modules are otherwise structured from a modularity standpoint. It might very well make sense to combine them as @iverase suggests.